frontend/public/components/utils/get-k8s-hook.ts

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { k8sGet, K8sKind, K8sResourceKind } from '../../module/k8s';
+
+export const useGetK8s = <R extends K8sResourceKind | K8sResourceKind[] = K8sResourceKind>(
```

Comment:
I'd expect this to always return a single resource and we'd have `useListK8s` for lists. I might call it `useK8sGet` to be consistent with `k8sGet`. We probably want `K8sResourceCommon` as the type here.

```suggestion
export const useK8sGet = <R extends K8sResourceCommon = K8sResourceKind>(
```

Reply:
`K8sResourceCommon` is probably better but its incompatible with `K8sResourceKind`.

`K8sResourceCommon` has all props required but `K8sResourceKind` has all props from `K8sResourceCommon` optional. It seems to me that `K8sResourceKind` should extend `K8sResourceCommon` (added to this PR) but I wonder if the `apiVersion`, `kind` and `metadata` should be optional or required ? So far I've marked them as optional.


Naming updated per your suggestion

Reply:
I think it makes sense for `K8sResourceKind` to extend `K8sResourceCommon`.

Should the default be `K8sResourceCommon`? I'd like to start getting away from `K8sResourceKind` since it doesn't guard against property name typos due to `[key: string]: any;`

https://github.com/openshift/console/blob/master/frontend/public/module/k8s/index.ts#L123

Reply:
makes sense, changed the default to `K8sResourceCommon`